tests/simplex/step-17.cc

```diff
@@ -0,0 +1,487 @@
+
+#include <fstream>
+#include <iostream>
+
+// simplex
+#include <deal.II/fe/mapping_fe.h>
+
+#include <deal.II/simplex/fe_lib.h>
+#include <deal.II/simplex/grid_generator.h>
+#include <deal.II/simplex/quadrature_lib.h>
+
+// Flag to toggle between hexes and simplices.
+// #define HEX
+
+namespace Step17
+{
+  using namespace dealii;
+
+  template <int dim>
+  class ElasticProblem
+  {
+  public:
+    ElasticProblem();
+    void
+    run();
+
+  private:
+    void
+    setup_system();
+
+    void
+    assemble_system();
+
+    unsigned int
+    solve();
+
+    void
+    output_results() const;
+
+    MPI_Comm mpi_communicator;
+
+    const unsigned int n_mpi_processes;
+    const unsigned int this_mpi_process;
+
+    ConditionalOStream pcout;
+
+#ifdef HEX
+    MappingQGeneric<dim, dim> mapping;
+#else
+    MappingFE<dim, dim> mapping;
+#endif
+
+    Triangulation<dim> triangulation;
+    FESystem<dim>      fe;
+    DoFHandler<dim>    dof_handler;
+
+    AffineConstraints<double> hanging_node_constraints;
+
+    PETScWrappers::MPI::SparseMatrix system_matrix;
+
+    PETScWrappers::MPI::Vector solution;
+    PETScWrappers::MPI::Vector system_rhs;
+  };
+
+
+  template <int dim>
+  class RightHandSide : public Function<dim>
+  {
+  public:
+    virtual void
+    vector_value(const Point<dim> &p, Vector<double> &values) const override
+    {
+      Assert(values.size() == dim, ExcDimensionMismatch(values.size(), dim));
+      Assert(dim >= 2, ExcInternalError());
+
+      Point<dim> point_1, point_2;
+      point_1(0) = 0.5;
+      point_2(0) = -0.5;
+
+      if (((p - point_1).norm_square() < 0.2 * 0.2) ||
+          ((p - point_2).norm_square() < 0.2 * 0.2))
+        values(0) = 1;
+      else
+        values(0) = 0;
+
+      if (p.square() < 0.2 * 0.2)
+        values(1) = 1;
+      else
+        values(1) = 0;
+    }
+
+    virtual void
+    vector_value_list(const std::vector<Point<dim>> &points,
+                      std::vector<Vector<double>> &  value_list) const override
+    {
+      const unsigned int n_points = points.size();
+
+      Assert(value_list.size() == n_points,
+             ExcDimensionMismatch(value_list.size(), n_points));
+
+      for (unsigned int p = 0; p < n_points; ++p)
+        RightHandSide<dim>::vector_value(points[p], value_list[p]);
+    }
+  };
+
+
+  template <int dim>
+  ElasticProblem<dim>::ElasticProblem()
+    : mpi_communicator(MPI_COMM_WORLD)
+    , n_mpi_processes(Utilities::MPI::n_mpi_processes(mpi_communicator))
+    , this_mpi_process(Utilities::MPI::this_mpi_process(mpi_communicator))
+    , pcout(std::cout, (this_mpi_process == 0))
+#ifdef HEX
+    , mapping(1)
+    , fe(FE_Q<dim>(1), dim)
+#else
+    , mapping(Simplex::FE_P<dim>(1))
+    , fe(Simplex::FE_P<dim>(1), dim)
+#endif
+    , dof_handler(triangulation)
+  {}
+
+
+  template <int dim>
+  void
+  ElasticProblem<dim>::setup_system()
+  {
+    GridTools::partition_triangulation(n_mpi_processes, triangulation);
```

Comment:
I'd bet that this is the cause of our problems - we default to using METIS here, but different versions of METIS produce significantly different output. Could you switch this to use `GridTools::partition_triangulation_zorder()` instead? That should produce the same output on all machines.

Reply:
Sure thing, I'll change it as you suggest. Thanks for pointing this out!

However, I'm not sure this is causing the error. The few lines of diff in the test log suggest that the test crashes altogether, rather than giving an output different from the provided one. Moreover, the serial test seems to be crashing as well... I think I'll try to get my hands on a different machine to try and reproduce the error.